homeassistant/components/vallox/fan.py

```diff
@@ -0,0 +1,167 @@
+
+    def __init__(self, name, client, state_proxy):
+        """Initialize the fan."""
+        self._name = name
+        self._client = client
+        self._state_proxy = state_proxy
+        self._available = False
+        self._state = None
+        self._fan_speed_home = None
+        self._fan_speed_away = None
+        self._fan_speed_boost = None
+
+        try:
+            self._client.set_settable_address(METRIC_KEY_MODE, int)
+
+        except AttributeError as attr_err:
+            _LOGGER.error("Error making mode override settable: %s", attr_err)
+
+    @property
+    def should_poll(self):
+        """Do not poll the device."""
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def available(self):
+        """Return if state is known."""
+        return self._available
+
+    @property
+    def is_on(self):
+        """Return if device is on."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        return {
+            ATTR_PROFILE_FAN_SPEED_HOME['description']: self._fan_speed_home,
+            ATTR_PROFILE_FAN_SPEED_AWAY['description']: self._fan_speed_away,
+            ATTR_PROFILE_FAN_SPEED_BOOST['description']: self._fan_speed_boost,
+        }
+
+    async def async_added_to_hass(self):
+        """Call to update."""
+        async_dispatcher_connect(self.hass, SIGNAL_VALLOX_STATE_UPDATE,
+                                 self._update_callback)
+
+    @callback
+    def _update_callback(self):
+        """Call update method."""
+        self.async_schedule_update_ha_state(True)
+
+    async def async_update(self):
+        """Fetch state from the device."""
+        try:
+            # Fetch if the whole device is in regular operation state.
+            mode = self._state_proxy.fetch_metric(METRIC_KEY_MODE)
+            if mode == 0:
+                self._state = True
+            else:
+                self._state = None
+
+            # Fetch the profile fan speeds.
+            self._fan_speed_home = int(self._state_proxy.fetch_metric(
+                ATTR_PROFILE_FAN_SPEED_HOME['metric_key']))
+            self._fan_speed_away = int(self._state_proxy.fetch_metric(
+                ATTR_PROFILE_FAN_SPEED_AWAY['metric_key']))
+            self._fan_speed_boost = int(self._state_proxy.fetch_metric(
+                ATTR_PROFILE_FAN_SPEED_BOOST['metric_key']))
+
+            self._available = True
+
+        except (IOError, KeyError) as err:
+            self._available = False
+            _LOGGER.error("Error updating fan. %s", err)
+
+    async def async_turn_on(self, speed: str = None, **kwargs) -> None:
+        """Turn the device on."""
+        _LOGGER.debug("Turn on: %s", speed)
+
+        # Case speed == None equals the GUI toggle switch being activated.
+        if speed is None:
```

Comment:
Invert this and make it a guard clause instead. Ie return if true. Then we can outdent below.